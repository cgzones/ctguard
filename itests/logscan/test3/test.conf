logscan {
  logfile "test.log" {
  }

  state_file = "test.state"
  output_kind = file
  output_path = "test.output"

  systemd_input = false
}
