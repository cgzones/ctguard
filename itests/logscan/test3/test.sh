#!/bin/sh

set -eu

cleanup () {
  rm -f test.log test.output test.pid test.state
}

cleanup

touch test.state

../../../ctguard-logscan --cfg-file test.conf --unittest --foreground &
pid=$!
echo "daemon running with pid ${pid}\n"

sleep 2

# test normale logs, to new file
echo "test" >> test.log
echo "test2" >> test.log

sleep 2

pid2=`pidof ctguard-logscan` || true
if [ "X${pid}" != "X${pid2}" ]; then
  echo "daemon not running anymore!!\n${pid} vs ${pid2}\nFAILURE!!\n\n";
  exit 1
fi

kill -INT ${pid}

sleep 1

diff -u test.output test.output.expected

cleanup

echo "\n\n\nsuccess!!!\n\n"
