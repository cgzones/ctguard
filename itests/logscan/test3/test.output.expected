 [daemon] : ctguard-logscan started
 [test.log] : !File coming alive
 [test.log] : !File got replaced
 [test.log] : test
 [test.log] : test2
 [daemon] : ctguard-logscan shutting down...
